linera-chain/src/manager.rs

```diff
@@ -394,14 +403,11 @@ impl ChainManager {
                 let value = HashedCertificateValue::new_validated(executed_block.clone());
                 if let Some(certificate) = lite_cert.with_value(value) {
                     self.locked = Some(certificate);
+                    self.locked_blobs = self.proposed_blobs.clone();
```

Comment:
Since they are _always_ set together, it would feel safer to put them in a single field, i.e. make `locked` type `Option<(Certificate, Vec<Blobs>)>` (or a map of blobs, if that's useful).
Then one can't accidentally assign only one without the other.

Reply:
As much as I like this idea, I think for consistency we should either have both `proposed` and `locked` work like this, or neither. And the problem is that if we put the proposed blobs together with the block proposal, that means we always have to request those two fields together in a chain info query for example. Which means we'll have a block proposal, which has blobs, _and_ the pending blobs.
It'll be a pain to keep this within the gRPC message limit 😅 so I don't think it works :/ 

What I can do is have a couple functions like `update_proposed` and `update_locked` that always update both, and use that everywhere. Doesn't prevent people from changing just one, but adds another layer of protection 🤔